parquet-variant/src/main/java/org/apache/parquet/variant/VariantUtil.java

```diff
@@ -0,0 +1,795 @@
+      }
+      default:
+        switch (typeInfo) {
+          case NULL:
+          case TRUE:
+          case FALSE:
+            return 1;
+          case INT8:
+            return 2;
+          case INT16:
+            return 3;
+          case INT32:
+          case DATE:
+          case FLOAT:
+            return 5;
+          case INT64:
+          case DOUBLE:
+          case TIMESTAMP:
+          case TIMESTAMP_NTZ:
+          case TIME:
+          case TIMESTAMP_NANOS:
+          case TIMESTAMP_NANOS_NTZ:
+            return 9;
+          case DECIMAL4:
+            return 6;
+          case DECIMAL8:
+            return 10;
+          case DECIMAL16:
+            return 18;
+          case BINARY:
+          case LONG_STR:
+            return 1 + U32_SIZE + readUnsigned(value, pos + 1, U32_SIZE);
+          case UUID:
+            return 1 + UUID_SIZE;
+          default:
+            throw new UnknownVariantTypeException(typeInfo);
+        }
+    }
+  }
+
+  private static IllegalArgumentException unexpectedType(Type type) {
+    return new IllegalArgumentException("Expected type to be " + type);
+  }
+
+  private static IllegalArgumentException unexpectedType(Type[] types) {
+    return new IllegalArgumentException("Expected type to be one of: " + Arrays.toString(types));
+  }
+
+  public static boolean getBoolean(ByteBuffer value) {
+    checkIndex(value.position(), value.limit());
+    int basicType = value.get(value.position()) & BASIC_TYPE_MASK;
+    int typeInfo = (value.get(value.position()) >> BASIC_TYPE_BITS) & PRIMITIVE_TYPE_MASK;
+    if (basicType != PRIMITIVE || (typeInfo != TRUE && typeInfo != FALSE)) {
+      throw unexpectedType(Type.BOOLEAN);
+    }
+    return typeInfo == TRUE;
+  }
+
+  /**
+   * Returns a long value from Variant value `value[pos...]`.
+   * It is only legal to call it if `getType` returns one of Type.BYTE, SHORT, INT, LONG,
+   * DATE, TIMESTAMP, TIMESTAMP_NTZ, TIME, TIMESTAMP_NANOS, TIMESTAMP_NANOS_NTZ.
+   * If the type is `DATE`, the return value is guaranteed to fit into an int and
+   * represents the number of days from the Unix epoch.
+   * If the type is `TIMESTAMP/TIMESTAMP_NTZ`, the return value represents the number of
+   * microseconds from the Unix epoch.
+   * If the type is `TIME`, the return value represents the number of microseconds since midnight.
+   * If the type is `TIMESTAMP_NANOS/TIMESTAMP_NANOS_NTZ`, the return value represents the number of
+   * nanoseconds from the Unix epoch.
+   * @param value The Variant value
+   * @return The long value
+   */
+  public static long getLong(ByteBuffer value) {
+    checkIndex(value.position(), value.limit());
+    int basicType = value.get(value.position()) & BASIC_TYPE_MASK;
+    int typeInfo = (value.get(value.position()) >> BASIC_TYPE_BITS) & PRIMITIVE_TYPE_MASK;
+    String exceptionMessage =
+        "Expect type to be one of: BYTE, SHORT, INT, LONG, TIMESTAMP, TIMESTAMP_NTZ, TIME, TIMESTAMP_NANOS, TIMESTAMP_NANOS_NTZ";
```

Comment:
There's no need for a local variable. This can either be in the `IllegalStateException` or should be a static constant.